shell/platform/embedder/embedder.cc

```diff
@@ -532,6 +533,83 @@ struct _FlutterPlatformMessageResponseHandle {
   fml::RefPtr<flutter::PlatformMessage> message;
 };
 
+struct LoadedElfDeleter {
+  void operator()(Dart_LoadedElf* elf) {
+    if (elf) {
+      ::Dart_UnloadELF(elf);
+    }
+  }
+};
+
+using UniqueLoadedElf = std::unique_ptr<Dart_LoadedElf, LoadedElfDeleter>;
+
+struct _FlutterEngineAOTData {
+  UniqueLoadedElf loaded_elf = nullptr;
+  const uint8_t* vm_snapshot_data = nullptr;
+  const uint8_t* vm_snapshot_instrs = nullptr;
+  const uint8_t* vm_isolate_data = nullptr;
+  const uint8_t* vm_isolate_instrs = nullptr;
+};
+
+FlutterEngineResult FlutterEngineCreateAOTData(
+    const FlutterEngineAOTDataSource* source,
```

Comment:
If it's const, and providing null is invalid, can't it be a `const &` rather than a `const *`?

Reply:
The embedder has a C API, so we can't use references unfortunately.

Reply:
🤦 I completely forgot references are C++, not C.